src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -297,34 +383,39 @@ private DRSAccessURL signGoogleUrl(Snapshot snapshot, DRSAccessMethod accessMeth
   }
 
   private DRSObject drsObjectFromFSFile(
-      FSFile fsFile, String snapshotId, AuthenticatedUserRequest authUser) {
-    DRSObject fileObject = makeCommonDrsObject(fsFile, snapshotId);
+      FSFile fsFile, Snapshot snapshot, AuthenticatedUserRequest authUser, boolean passportAuth) {
+    DRSObject fileObject = makeCommonDrsObject(fsFile, snapshot.getId().toString());
 
     List<DRSAccessMethod> accessMethods;
     CloudPlatformWrapper platform = CloudPlatformWrapper.of(fsFile.getCloudPlatform());
     if (platform.isGcp()) {
-      accessMethods = getDrsAccessMethodsOnGcp(snapshotId, fsFile, authUser);
+      String gcpRegion = retrieveGCPSnapshotRegion(snapshot, fsFile);
+      if (passportAuth) {
+        accessMethods = getDrsSignedURLAccessMethods("gcp-passport-", gcpRegion);
```

Comment:
What do you think about making these prefixes constants?

Reply:
Good point! Should be constants now.